mlflow/utils/conda.py

```diff
@@ -133,10 +191,18 @@ def get_or_create_conda_env(conda_env_path, env_id=None, capture_output=False):
             )
         )
 
-    env_names = _list_conda_environments()
-    project_env_name = _get_conda_env_name(conda_env_path, env_id)
-    if project_env_name not in env_names:
-        _logger.info("=== Creating conda environment %s ===", project_env_name)
+    conda_extra_env_vars = _get_conda_extra_env_vars(env_root_dir)
+
+    # Include the env_root_dir hash in the project_env_name,
+    # this is for avoid conda env name conflicts between different CONDA_ENVS_PATH.
+    project_env_name = _get_conda_env_name(conda_env_path, env_id=env_id, env_root_dir=env_root_dir)
+    if env_root_dir is not None:
+        project_env_path = os.path.join(env_root_dir, _CONDA_ENVS_DIR, project_env_name)
+    else:
+        project_env_path = project_env_name
+
+    if project_env_name not in _list_conda_environments(conda_extra_env_vars):
+        _logger.info("=== Creating conda environment %s ===", project_env_path)
```

Comment:
```suggestion
        _logger.info("=== Creating conda environment %s ===", project_env_name)
```

Can we use `project_env_name`?

Reply:
I prefer print full path because print full path helps debugging.

Reply:
Ok